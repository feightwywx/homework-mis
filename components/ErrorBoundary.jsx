import React from "react";

export class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error) {
    console.error(error)
    // 更新 state 使下一次渲染能够显示降级后的 UI
    return { hasError: true };
  }

  componentDidCatch(error, errorInfo) {
    // 你同样可以将错误日志上报给服务器
    console.error(error, errorInfo);
  }

  render() {
    return this.props.children; 
  }
}