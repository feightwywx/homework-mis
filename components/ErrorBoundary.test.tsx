import { render } from '@testing-library/react'
import { ErrorBoundary } from './ErrorBoundary'

describe('ErrorBoundary', () => {
  it('renders ErrorBoundary', () => {
    const { container } = render(<ErrorBoundary />);
    expect(container).toMatchSnapshot();
  })
})