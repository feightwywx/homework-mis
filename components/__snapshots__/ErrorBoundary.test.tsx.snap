// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary renders ErrorBoundary 1`] = `<div />`;
